### PR TITLE
Better fail message

### PR DESCRIPTION
**Pull request by Rlamboll**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <https://github.com/znicholls/silicone/pull/XX>`_) Added feature which does something
- (`#XX <https://github.com/znicholls/silicone/pull/XX>`_) Fixed bug identified in (`#YY <https://github.com/znicholls/silicone/issues/YY>`_)
```

### CODE REVIEW

**Comment by Rlamboll** [comment]
We can do, although currently there are rather a lot of reasons why it may fail with a ValueError. This sort of warning will still need to be caught anyway. Do we want to ignore all ValueErrors from a single run of silicone in those cases? Quite possibly we do, I suppose.

**Comment by znicholls** [comment]
> Do we want to ignore all ValueErrors from a single run of silicone in those cases?

So I think given this PR is just about what happens if there's no data, let's think about that first. As a user I'd expect an error with the message you have. 

Then when we're running Silicone elsewhere (e.g. AR6 stuff) we can put the error handling in there. However in this PR I think an error is still most appropriate. If we wanted to get really fancy, we could create our own ValueError sub-class, NoData, or something.

**Comment by Rlamboll** [comment]
I've changed the error message, however I still get this weird tagging problem that means that half of the notepad checks fail. Am I just using a different package for something?

**Comment by znicholls** [comment]
> Am I just using a different package for something?

No the order of columns in dataframes isn't stable. Just put `# NBVAL_IGNORE_OUTPUT` at the top of each of the failing cells.

**Comment by byersiiasa** [comment]
Thanks a lot @Rlamboll @znicholls  - this is useful - so we will use a try/except block - but at least the warning will indicate why the particular scenario fails.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -3,6 +3,7 @@ Changelog
 
 master
 ------
+- (`#32 <https://github.com/znicholls/silicone/pull/322>`_) Made asking to infill a case with no data give an explicit warning
```
Comment (znicholls) [inline CHANGELOG.rst]:
```suggestion
- (`#32 <https://github.com/znicholls/silicone/pull/322>`_) Raise `ValueError` when asking to infill a case with no data
```

**Pull request merged**